pysph/sph/wc/crksph.py

```diff
@@ -373,3 +384,468 @@ def loop(self, d_idx, s_idx, d_ai, d_gradai, d_cwij, d_bi, d_gradbi, s_ai,
+        s_{\max} / (s_{\min} + s_{\max}) &\Delta u_{ij}\times(s_i - s_j) < 0\\
+        \end{cases}
+
+    .. math::
+        s_{\min} = \min(|s_i|, |s_j|)
+
+    .. math::
+        s_{\max} = \max(|s_i|, |s_j|)
+
+    .. math::
+        s_{i/j} = \frac{p_{i/j}}{\rho_{i/j}^\gamma}
+
+    see MomentumEquation for :math:`\frac{Dv_{ij}^{\alpha}}{Dt}`
+    """
+
+    def __init__(self, dest, sources, dim, gamma):
+        self.dim = dim
+        self.gamma = gamma
+        super(EnergyEquation, self).__init__(dest, sources)
+
+    def _get_helpers_(self):
+        return [dot]
+
+    def initialize(self, d_idx, d_ae):
+        d_ae[d_idx] = 0.0
+
+    def loop(self, d_idx, s_idx, d_au, d_av, d_aw, d_ae, s_au, s_av, s_aw,
+             d_u0, d_v0, d_w0, s_u0, s_v0, s_w0, d_u, d_v, d_w, s_u, s_v, s_w,
+             d_p, d_rho, s_p, s_rho):
+        gamma = self.gamma
+        d = declare('int')
+        d = self.dim
+        auij, delu = declare('matrix(3)', 2)
+        auij[0] = d_au[d_idx] - s_au[s_idx]
+        auij[1] = d_av[d_idx] - s_av[s_idx]
+        auij[2] = d_aw[d_idx] - s_aw[s_idx]
+
+        delu[0] = s_u0[s_idx] + s_u[s_idx] - d_u0[d_idx] - d_u[d_idx]
+        delu[1] = s_v0[s_idx] + s_v[s_idx] - d_v0[d_idx] - d_v[d_idx]
+        delu[2] = s_w0[s_idx] + s_w[s_idx] - d_w0[d_idx] - d_w[d_idx]
+
+        aeij = dot(delu, auij, d)
+
+        si = d_p[d_idx]/((d_rho[d_idx])**gamma)
+        sj = s_p[s_idx]/((s_rho[s_idx])**gamma)
+        smin = min(abs(si), abs(sj))
+        smax = max(abs(si), abs(sj))
+
+        fij = 0.5
+        sdiff = si - sj
+        if sdiff * aeij > 0:
+            fij = smin/(smin + smax)
+        elif sdiff * aeij < 0:
+            fij = smax/(smin + smax)
+
+        d_ae[d_idx] += 0.5*fij * aeij
+
+
+class StateEquation(Equation):
+    def __init__(self, dest, sources, gamma):
+        self.gamma = gamma
+        super(StateEquation, self).__init__(dest, sources)
+
+    def initialize(self, d_idx, d_p, d_rho, d_e):
+        d_p[d_idx] = (self.gamma - 1) * d_rho[d_idx] * d_e[d_idx]
+
+
+def get_particle_array_crksph(constants=None, **props):
+    crksph_props = [
+        'e', 'au', 'av', 'aw', 'ae', 'u0', 'v0', 'w0', 'cs', 'V', 'rhofac',
+        'x0', 'y0', 'z0', 'rho0', 'ax', 'ay', 'az', 'arho'
+    ]
+
+    pa = get_particle_array(
+        additional_props=crksph_props, constants=constants, **props
+    )
+    pa.add_property('cwij')
+    pa.add_property('ai')
+    pa.add_property('bi', stride=3)
+    pa.add_property('gradai', stride=3)
+    pa.add_property('gradbi', stride=9)
+    pa.add_property('gradv', stride=9)
+    pa.add_output_arrays(['p', 'V'])
+    return pa
+
+
+class CRKSPHScheme(Scheme):
+    def __init__(self, fluids, solids, dim, rho0, c0, nu, h0, p0, gx=0.0,
+                 gy=0.0, gz=0.0, cl=2, cq=1, gamma=7.0, eta_crit=0.5,
+                 eta_fold=0.2, tol=0.5):
+        self.fluids = fluids
+        self.solids = solids
```

Comment:
I don't think this deals with solids at all, so why not just remove the solids argument for now instead of people using it and then getting wrong results?

Reply:
removed it.